### ISSUE TITLE
PEP avatars support (XEP-0084)

### ISSUE THREAD

**Issue by moshpirit** [redsolution/xabber-android#459]
Add [XEP-0084](https://xmpp.org/extensions/xep-0084.html) for more avatars support.

Please, read [this Conversations issue for more information about it](https://github.com/siacs/Conversations/issues/69), specially [this message](https://github.com/siacs/Conversations/issues/69#issuecomment-49864163):

>I like to have pictures of my contacts. In fact this is pretty much the reason Conversations had contact list integration from day one.
I see that adding every jabber contact to your address book might not be a viable option. So I had another, closer, look at the XEPs.

>XEP-0153 is the most widely adopted. Unfortunately it's complete garbage. It's not even a proper XEP (as in proposed to become a standard). It's just a write down of what legacy clients did in the past. Images can only be stored in base64. It limits the size to 8KB (which after base64 becomes 32KB + vcard overhead (one always has to download the entire vcard)). The image size is recommended to be 64px and should max out at 96px.
The avatars in Conversations (in the Conversations overview) are, depending on your phones dpi 128px or even larger. So this will basically look like crap.
One has to fetch ones own vcard/avater on every login which on mobile is without stream management can become very very traffic intensive.

>XEP-0008 shares a lot of problems with XEP-0153. The only benefit is that one doesn't have the vcard overhead (which isn't that huge). Plus the adoption rate is lower afaik.

>XEP-0084 does a very good job on separating meta data and the actual images. One doesn't have to download ones image on every login but just check the meta data. XEP-0084 can handle multiple images in multiple formats and resolutions. So we could offer a general purpose JPEG image in a generic solution (lets say 96px) and a higher res webp image (which is awesome at high compression rates) especially for Conversations. Plus XEP-0084 allows images to be stored on a HTTP server as well. This could work around the base64 problem. (However this is optional)

>On the downside the adoption rate of XEP-0084 is lower and it requires the server to support PEP. But a low adoption rate hasn't botherd Conversations in the past if it was the better solution.

>**TLDR**: XEP 0084 is pretty much the only reasonable one to use. In contrary to what i said before PEP (as opposed to pupsub) is actually designed to work with IM.
